### PR TITLE
Eliminates older CM API code + replace with the better feature detection

### PR DESCRIPTION
**Pull request by agektmr**

### CODE REVIEW

**Comment by agektmr** [comment]
cc: @engedy

File: static/scripts/app.js [hunk 1]
```diff
@@ -112,9 +91,9 @@ app._fetch = async function(provider, c = new FormData()) {
  * @return {Promise} Resolves if credential info is available.
  */
 app._autoSignIn = async function(silent) {
-  if (app.cmaEnabled) {
+  if (window.PasswordCredential) {
```
Comment (engedy) [inline static/scripts/app.js]: Technically this is OK, but semantically, the intention seems to be to retrieve federated credentials as well, so maybe we should test for window.FederatedCredential too?

File: static/scripts/app.js [hunk 2]
```diff
@@ -388,12 +359,12 @@ app.onRegister = function(e) {
...
 app.onUnregister = function() {
   // POST `id` to `/unregister` to unregister the user
-  let form = new FormData();
+  const form = new FormData();
   form.append('id', app.userProfile.id);
 
   app._fetch(UNREGISTER, form)
   .then(() => {
-    if (app.cmaEnabled) {
+    if (window.PasswordCredential) {
```
Comment (engedy) [inline static/scripts/app.js]: Should we instead test for preventSilentAccess here?

File: static/scripts/app.js [hunk 3]
```diff
@@ -418,7 +389,7 @@ app.onUnregister = function() {
 app.signOut = function() {
   app._fetch(SIGNOUT)
   .then(() => {
-    if (app.cmaEnabled) {
+    if (window.PasswordCredential) {
```
Comment (engedy) [inline static/scripts/app.js]: Same goes here.

**Pull request closed without merging**